source/common/stats/utility.h

```diff
@@ -12,6 +13,29 @@
 namespace Envoy {
 namespace Stats {
 
+/**
+ * Represents a dynamically created stat name token based on absl::string_view.
+ * This class wrapper is used in the 'Element' variant so that call-sites
+ * can express explicit intent to create dynamic stat names, which are more
+ * expensive than symbolic stat names. We use dynamic stat names only for
+ * building stats based on names discovered in the line of a request.
+ */
+class DynamicName : public absl::string_view {
+public:
+  DynamicName(absl::string_view s) : absl::string_view(s) {}
```

Comment:
can you do `using absl::string_view::string_view` instead to just bring the parent constructors into scope? Another thought: should we require an explicit conversion? I think currently passing an `absl::string_view` to `Element` would get implicitly converted?

Reply:
I want the conversion to be implicit as it makes the usage model much nice. I added comments.

I also switched to using `using` to pull in the parent-class ctor. I haven't typically done that, but it seems fine too and is a littler terser.